paper/VIT/train_vit.py

```diff
@@ -0,0 +1,225 @@
+import argparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision
+import pytorch_lightning as pl
+import warmup_scheduler
+import numpy as np
+from utils import get_model, get_dataset, get_experiment_name, get_criterion
+from da import CutMix, MixUp
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--api-key", help="API Key for Comet.ml")
+parser.add_argument("--dataset", default="c10", type=str, help="[c10, c100, svhn]")
+parser.add_argument("--num-classes", default=10, type=int)
+parser.add_argument("--model-name", default="vit", help="[vit]", type=str)
+parser.add_argument("--patch", default=8, type=int)
+parser.add_argument("--batch-size", default=128, type=int)
+parser.add_argument("--eval-batch-size", default=1024, type=int)
+parser.add_argument("--lr", default=1e-3, type=float)
+parser.add_argument("--min-lr", default=1e-5, type=float)
+parser.add_argument("--beta1", default=0.9, type=float)
+parser.add_argument("--beta2", default=0.999, type=float)
+parser.add_argument("--off-benchmark", action="store_true")
+parser.add_argument("--max-epochs", default=200, type=int)
+parser.add_argument("--dry-run", action="store_true")
+parser.add_argument("--weight-decay", default=5e-5, type=float)
+parser.add_argument("--warmup-epoch", default=5, type=int)
+parser.add_argument("--precision", default=16, type=int)
+parser.add_argument("--autoaugment", action="store_true")
+parser.add_argument("--criterion", default="ce")
+parser.add_argument("--label-smoothing", action="store_true")
+parser.add_argument("--smoothing", default=0.1, type=float)
+parser.add_argument("--rcpaste", action="store_true")
+parser.add_argument("--cutmix", action="store_true")
+parser.add_argument("--mixup", action="store_true")
+parser.add_argument("--dropout", default=0.0, type=float)
+parser.add_argument("--head", default=12, type=int)
+parser.add_argument("--num-layers", default=7, type=int)
+parser.add_argument("--hidden", default=384, type=int)
+parser.add_argument("--mlp-hidden", default=384, type=int)
+parser.add_argument("--off-cls-token", action="store_true")
+parser.add_argument("--seed", default=42, type=int)
+parser.add_argument("--project-name", default="VisionTransformer")
+parser.add_argument("--data_samples", default=1000, type=int)
+parser.add_argument("--mem_samples", default=100, type=int)
+parser.add_argument("--data_root", default="data", type=str)
+args = parser.parse_args()
+torch.manual_seed(args.seed)
+np.random.seed(args.seed)
+args.benchmark = True 
+args.gpus = torch.cuda.device_count()
+args.num_workers = 4*args.gpus if args.gpus else 8
+args.is_cls_token = True if not args.off_cls_token else False
+
+
+import numpy as np
+import random
+import os 
+from typing import List
+def split_dataset(dataset:torch.utils.data.Dataset,train_size:int,val_size:int,seed:int)->List[torch.utils.data.Dataset]:
```

Comment:
Space is missing. Two lines of space are the general rule

Reply:
Fixed in [092868b](https://github.com/KRLGroup/memory-wrap/pull/3/commits/092868bd193ae604b4f113c743556c790a4eabbc)